### PR TITLE
MAISTRA-1399 default to NetworkPolicy if an unknown network type is in use

### PR DESCRIPTION
**Pull request by rcernich**
This simply removes the allowlist check and defaults to NetworkPolicy.

Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

**Comment by dgn** [comment]
I guess LGTM is right; in line 105 we're introducing another `err` var that is shadowing the one from func scope

File: pkg/controller/servicemesh/memberroll/namespace_reconciler.go [hunk 1]
```diff
@@ -121,7 +121,8 @@ func (r *namespaceReconciler) initializeNetworkingStrategy(ctx context.Context)
 						return fmt.Errorf("unsupported cluster network plugin: %s", networkPlugin)
 					}
 				} else {
-					log.Info("cluster network plugin not defined, skipping network configuration")
+					log.Info("cluster network plugin not defined, defaulting to NetworkPolicy", networkType)
+					r.networkingStrategy, err = newNetworkPolicyStrategy(ctx, r.Client, r.meshNamespace)
```
Comment (dgn) [inline pkg/controller/servicemesh/memberroll/namespace_reconciler.go:126]: if we return `err` after this the alert will go away

File: pkg/controller/servicemesh/memberroll/namespace_reconciler.go [hunk 2]
```diff
@@ -121,7 +121,8 @@ func (r *namespaceReconciler) initializeNetworkingStrategy(ctx context.Context)
 						return fmt.Errorf("unsupported cluster network plugin: %s", networkPlugin)
 					}
 				} else {
-					log.Info("cluster network plugin not defined, skipping network configuration")
+					log.Info("cluster network plugin not defined, defaulting to NetworkPolicy", networkType)
```
Comment (luksa) [inline pkg/controller/servicemesh/memberroll/namespace_reconciler.go]:
```suggestion
					log.Info("cluster network plugin not defined, defaulting to NetworkPolicy", "networkType", networkType)
```

File: pkg/controller/servicemesh/memberroll/namespace_reconciler.go [hunk 3]
```diff
@@ -130,13 +131,16 @@ func (r *namespaceReconciler) initializeNetworkingStrategy(ctx context.Context)
 				log.Info("Network Strategy OVNKubernetes:NetworkPolicy")
 				r.networkingStrategy, err = newNetworkPolicyStrategy(ctx, r.Client, r.meshNamespace)
 			default:
-				return fmt.Errorf("unsupported network type: %s", networkType)
+				log.Info("Unknown Network Type %v, defaulting to NetworkPolicy", networkType)
```
Comment (luksa) [inline pkg/controller/servicemesh/memberroll/namespace_reconciler.go]:
```suggestion
				log.Info("Unknown Network Type, defaulting to NetworkPolicy", "networkType", networkType)
```

Comment (rcernich) [reply pkg/controller/servicemesh/memberroll/namespace_reconciler.go]: fixed.

File: pkg/controller/servicemesh/memberroll/namespace_reconciler.go (hunk 2 above)
Comment (rcernich) [reply pkg/controller/servicemesh/memberroll/namespace_reconciler.go]: fixed

File: pkg/controller/servicemesh/memberroll/namespace_reconciler.go (hunk 1 above)
Comment (rcernich) [reply pkg/controller/servicemesh/memberroll/namespace_reconciler.go:126]: fixed

**Review by luksa — APPROVED** [review]

**Review by jwendell — APPROVED** [review]

**Pull request merged**